psutil/tests/test_posix.py

```diff
@@ -314,19 +314,27 @@ def test_os_waitpid_bad_ret_status(self):
 
     def test_disk_usage(self):
         def df(device):
-            out = sh("df -B 1 %s" % device).strip()
+            # Use 1 kB block sizes since OS X doesn't have -B flag
+            out = sh("df -k 1 %s" % device).strip()
             line = out.split('\n')[1]
             fields = line.split()
-            total = int(fields[1])
-            used = int(fields[2])
-            free = int(fields[3])
+            total = int(fields[1]) * 1024
+            used = int(fields[2]) * 1024
+            free = int(fields[3]) * 1024
             percent = float(fields[4].replace('%', ''))
             return (total, used, free, percent)
 
         tolerance = 4 * 1024 * 1024  # 4MB
         for part in psutil.disk_partitions(all=False):
             usage = psutil.disk_usage(part.mountpoint)
-            total, used, free, percent = df(part.device)
+            try:
+                total, used, free, percent = df(part.device)
+            except RuntimeError as err:
+                # Issue on Mac OS X and Linux not being able to read certain
+                # locations such as /dev/mapper/docker* so skip them
+                if ('/dev/mapper/docker' in str(err) and
+                    'No such file or directory' in str(err)):
+                    continue
```

Comment:
Add an `else:` here and indent the block below so that you'll fix `UnboundLocalError: local variable 'total' referenced before assignment` error.